indexer/editable_map_object.cpp

```diff
@@ -280,6 +280,14 @@ void EditableMapObject::SetWebsite(string website)
 void EditableMapObject::SetInternet(Internet internet)
 {
   m_metadata.Set(feature::Metadata::FMD_INTERNET, DebugPrint(internet));
+
+  if (internet == Internet::Unknown)
+  {
+    static const uint32_t wifiType = classif().GetTypeByPath({"internet_access", "wlan"});
```

Comment:
It looks like a false optimization. All statics are guarded by memory
barriers - this guarantees correct initialization and usage of statics in
multithreaded environment. It's generally OK for singletons, but is not
needed for such simple things. Please, don't use statics without real need.

18 авг. 2016 г. 18:33 пользователь "Arsentiy Milchakov" <
notifications@github.com> написал:

> In indexer/editable_map_object.cpp
> <https://github.com/mapsme/omim/pull/4080#discussion_r75331484>:
>
> > @@ -280,6 +280,14 @@ void EditableMapObject::SetWebsite(string website)
> >  void EditableMapObject::SetInternet(Internet internet)
> >  {
> >    m_metadata.Set(feature::Metadata::FMD_INTERNET, DebugPrint(internet));
> > +
> > +  if (internet == Internet::Unknown)
> > +  {
> > +    static const uint32_t wifiType = classif().GetTypeByPath({"internet_access", "wlan"});
>
> Для того чтобы каждый раз не пересоздавать переменную, чтобы работало
> быстрее. Тип вайфая не меняется в процессе выполнения программы.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/mapsme/omim/pull/4080/files/b280e24da30bd32269022090acdf4c51a93f8733#r75331484>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAbvp_jFNjB3HwAZxOniIOgIZPykgYPBks5qhHs8gaJpZM4JmdzA>
> .
>